src/providers/generalWalletProvider.tsx

```diff
@@ -314,6 +324,50 @@ export const GeneralWalletProvider: React.FC<{
 		}
 	};
 
+	const sendSolanaSPLToken = async (
+		to: string,
+		value: bigint,
+		tokenAddress: string,
+	) => {
+		if (!publicKey) throw Error('Wallet is not connected');
+		const splTokenMintAddress = new PublicKey(tokenAddress);
+		const receiverAddress = new PublicKey(to);
+		const senderTokenAccountAddress = await getAssociatedTokenAddress(
+			splTokenMintAddress,
+			publicKey,
+		);
+		const receiverTokenAccountAddress = await getAssociatedTokenAddress(
+			splTokenMintAddress,
+			receiverAddress,
+		);
+		const transaction = new Transaction();
+		const receiverAccountInfo = await solanaConnection.getAccountInfo(
+			receiverTokenAccountAddress,
+		);
+		if (!receiverAccountInfo) {
+			// In the case where user is new to the token and doesn't have an associated token account
+			transaction.add(
+				createAssociatedTokenAccountInstruction(
+					publicKey,
+					receiverTokenAccountAddress,
+					receiverAddress,
+					splTokenMintAddress,
+				),
+			);
+		}
+		transaction.add(
+			createTransferInstruction(
+				senderTokenAccountAddress,
+				receiverTokenAccountAddress,
+				publicKey,
+				value,
+				[],
+				TOKEN_PROGRAM_ID,
+			),
+		);
+		return await solanaSendTransaction(transaction, solanaConnection);
+	};
```

Comment:
Please move this method to src/hooks/useCreateSolanaDonation.tsx since it's the only place it's used and it's not needed to cluter the `generalWalletProvider`

Reply:
Done!